### ISSUE TITLE
What's purpose of flipping query_labels?

### ISSUE THREAD

**Issue by Leonardyao** [cfifty/CAML#4]
Hello,

I'm confused about the operation to flip query labels(code: `query_labels = torch.flip(query_labels, dims=[0])`) in training loop, could you please explain why? 

Thank you

**Comment by cfifty** [comment]
Much of the evaluation code (i.e. downstream dataset evaluation) is based on FRN [https://github.com/Tsingularity/FRN]. Flipping aligns the order of query labels in training with the order of query labels in eval. 

Recall labels in meta-learning benchmarks are somewhat arbitrary: 

Setting the labels as (cow -> 1), (dog -> 2), (cat -> 3)  should be equivalent to the assignment of (cow -> 3), (dog -> 1), (cat ->2) since labels are *relative* (not absolute like typical image classification datasets where 0 might always correspond to the "dog" label). 

Specifically, if the model was trained to predict "class 5", we don't want the label to switch to "class 1" during down-stream evaluation. 

You can look at the way support-set and query-set labels are constructed in training vs. down-stream evaluation to verify this.
